streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -310,18 +313,14 @@ public void enforceRestoreActive() {
     // be cleared when the corresponding task is being removed from the thread. In other words, the restore consumer
     // should contain all changelogs that are RESTORING or COMPLETED
     @Override
-    public void transitToUpdateStandby() {
+    public synchronized void transitToUpdateStandby() {
```

Comment:
I'm also wondering if it even makes sense to do just the restore in a separate thread and not also move the standbys, assuming we do indeed want to move standbys to a separate thread eventually. It just seems like we might need to do a lot of work to get things consistent in the interim that would all be thrown out when we move on to standbys.

Of course, if we don't plan to move standbys to a separate thread anytime soon then there's no problem. Just wondering if that's part of the near-future roadmap (since standby improvements do indirectly translate to restoration improvements)

Reply:
The ultimate plan, which is also the goal of this PR, is to move both processing of standbys / restoring of actives, which are both captured in the ChangelogReader, to the same separate thread out of the main thread. Do you see a concern with this plan?